src/app/core/components/left-nav-tree/left-nav-tree.component.ts

```diff
@@ -20,20 +31,40 @@ type LeftNavTreeNode = TreeNode<{
   templateUrl: './left-nav-tree.component.html',
   styleUrls: [ './left-nav-tree.component.scss' ]
 })
-export class LeftNavTreeComponent implements OnChanges {
+export class LeftNavTreeComponent implements OnInit, OnChanges, OnDestroy {
   @Input() data?: NavTreeData<NavTreeElement>;
   @Input() elementType: ItemTypeCategory | 'group' = 'activity';
 
+  subscription?: Subscription;
+  selectNodeIdEvent = new EventEmitter<string>();
+  componentStateChange = new EventEmitter<void>();
   nodes: LeftNavTreeNode[] = [];
 
   constructor(
     private itemRouter: ItemRouter,
     private groupRouter: GroupRouter,
     private router: Router,
+    private leftNavScrollService: LeftNavScrollService,
   ) {}
 
+  ngOnInit(): void {
+    this.subscription = combineLatest([
+      this.selectNodeIdEvent.asObservable(),
+      this.componentStateChange.asObservable(),
+    ]).pipe(
+      debounceTime(500),
+    ).subscribe(([ id ]) => {
+      this.leftNavScrollService.scrollToMenuItemForId(id);
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscription?.unsubscribe();
+  }
+
   ngOnChanges(_changes: SimpleChanges): void {
     this.nodes = this.data ? this.mapItemToNodes(this.data) : [];
+    this.componentStateChange.emit();
```

Comment:
`this.data` (and so `mapItemToNodes`) already contains the information about the selected node... so I don't see the point of line 121 `this.selectNodeIdEvent.emit(node.data.element.id);` which store this information a second time.

(related with my previous comment)
Couldn't you even do everything in left-nav directly? Around line 50, you know that the id has changed, so you could directly tell left-menu (the parent component) to scroll.

Reply:
Fixed